proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,398 @@
+                           Expression,
+                           Constraint,
+                           value,
+                           units as pyunits,
+                           SolverStatus)
+from pyomo.util.check_units import assert_units_consistent, check_units_equivalent
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import (degrees_of_freedom,
+                                              number_variables,
+                                              number_total_constraints,
+                                              number_unused_variables)
+from idaes.core.util.scaling import (calculate_scaling_factors,
+                                     unscaled_variables_generator,
+                                     unscaled_constraints_generator,
+                                     badly_scaled_var_generator)
+from idaes.core.util.exceptions import (PropertyPackageError,
+                                        PropertyNotSupportedError)
+from idaes.core.util import get_solver
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+solver.options["nlp_scaling_method"] = "user-scaling"
+
+
+# -----------------------------------------------------------------------------
+class PropertyUnitTestHarness(object):
+    def configure_class(self, m):
+        self.prop_pack = None
+        self.param_args = {}
+        self.set_default_scaling_dict = {}  # keys = (var string, tuple index), value = scaling factor
+
+        self.component_dict = {}  # keys = component string, value = component type
+        self.phase_dict = {}  # keys = phase string, value = phase type
+        self.param_obj_dict = {}  # keys = pyomo object string, value = (type, units)
+        self.default_scaling_factor_dict = {}  # keys = (var string, index), value = scaling factor
+
+        self.state_vars_list = None
+        self.stateblock_obj_dict = {}  # keys = pyomo object string, value = (type, units)
+
+        self.statistics_dict = {}  # keys = statistic string, value = number
+        self.general_methods_dict = {}  # keys = (method name string, tuple index), value = (var, tuple index)
+        self.default_methods_dict = {}  # keys = method name string, value = class attribute
+
+        self.configure()
+
+        # attaching objects to model
+        m.component_dict = self.component_dict
+        m.phase_dict = self.phase_dict
+        m.param_obj_dict = self.param_obj_dict
+        m.default_scaling_factor_dict = self.default_scaling_factor_dict
+
+        m.stateblock_obj_dict = self.stateblock_obj_dict
+        m.state_vars_list = self.state_vars_list
+
+        m.statistics_dict = self.statistics_dict
+        m.general_methods_dict = self.general_methods_dict
+        m.default_methods_dict = self.default_methods_dict
+
+
+    def configure(self):
+        # Placeholder method to allow user to setup test harness
```

Comment:
Need to document what is required here.

Reply:
See comment above about following IDAES API for the property test harness.

Reply:
I added doc strings for all the requirements.